launch/launch.go

```diff
@@ -31,6 +39,64 @@ type Metadata struct {
 	Buildpacks []Buildpack `toml:"buildpacks" json:"buildpacks"`
 }
 
+// DecodeLaunchMetadataTOML reads a launch.toml file
+func DecodeLaunchMetadataTOML(path string, platformAPI *api.Version, launchmd *Metadata) error {
+	// decode the common bits
+	md, err := toml.DecodeFile(path, &launchmd)
+	if err != nil {
+		return err
+	}
+
+	// decode the process.commands, which will differ based on platform API
+	commandsAreStrings := true
+
+	// processes are defined differently depending on API version
+	// and will be decoded into different values
+	for i, process := range launchmd.Processes {
+		if commandsAreStrings {
+			var commandString string
+			if err = md.PrimitiveDecode(process.RawCommandValue, &commandString); err != nil {
+				return err
+			}
+
+			launchmd.Processes[i].Command = []string{commandString}
+		} else {
+			var command []string
+			if err = md.PrimitiveDecode(process.RawCommandValue, &command); err != nil {
+				return err
+			}
+			launchmd.Processes[i].Command = command
+		}
+	}
+
+	return nil
+}
+
+func (m Metadata) Matches(x interface{}) bool {
```

Comment:
What's the code path where this is used? Is it for tests? A comment here could be helpful for posterity...